### PR TITLE
integrations/operator: Add openSSH and openSSHEICE servers

### LINKED ISSUE #36839: Support declaring agentless nodes via the operator
### What would you like Teleport to do?

The Teleport operator should support creating agentless openssh nodes like in https://goteleport.com/docs/server-access/guides/openssh/openssh-manual-install/

### What problem does this solve?

User choosing the operator to control Teleport must be able to declare their agentless nodes the same way they manage other resources.

### If a workaround exists, please include it.

Handle those nodes via `tctl create` instead.

### PR DESCRIPTION
**Pull request by hugoShaka**
Fixes: https://github.com/gravitational/teleport/issues/36839

This PR adds OpenSSH and OpenSSHEICE support to the operator through two  new resources:

- `TeleportOpensshServer`
- `TeleportOpensshEICEServer`

Notes about the 1 resource vs many resources for the subkinds:
- `teleport`, `openssh`, and `openssh-aws-eice` subkinds are backed by the same resource
- the operator must not manage the `teleport` subkind
- there is no Kubernetes-friendly way to pass a subkind, putting it in the spec would break the unmarshalling

changelog: Add `OpensshServer` and `OpensshEICEServer` support in the operator

### CODE REVIEW

File: examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_openssheiceserversv2.yaml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                        type: string
+                      vpc_id:
+                        description: VPCID is the AWS VPC ID where the Instance is
+                          running.
+                        type: string
+                    type: object
+                type: object
+              cmd_labels:
+                description: CmdLabels is server dynamic labels
+                properties:
```
Comment (hugoShaka) [inline examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_openssheiceserversv2.yaml:80]: TODO: hide this field, this must not be set by the user.

File: integrations/operator/controllers/resources/global.go [hunk 2]
```diff
@@ -75,6 +75,8 @@ func SetupAllControllers(log logr.Logger, mgr manager.Manager, teleportClient *c
 		{"TeleportGithubConnector", NewGithubConnectorReconciler},
 		{"TeleportProvisionToken", NewProvisionTokenReconciler},
 		{"TeleportOktaImportRule", NewOktaImportRuleReconciler},
+		{"TeleportOpensshServerV2", NewOpensshServerV2Reconciler},
+		{"TeleportOpensshEICEServerV2", NewOpensshEICEServerV2Reconciler},
```
Comment (marcoandredinis) [inline integrations/operator/controllers/resources/global.go:79]: Can we use the correct case for `OpenSSH`?

File: integrations/operator/controllers/resources/openssheiceserverv2_controller.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// Get gets the Teleport provision token of a given name
+func (r opensshEICEServerClient) Get(ctx context.Context, name string) (types.Server, error) {
+	server, err := r.teleportClient.GetNode(ctx, defaults.Namespace, name)
+	if err != nil {
+		return server, trace.Wrap(err)
+	}
+	if subKind := server.GetSubKind(); subKind != types.SubKindOpenSSHEICENode {
+		return nil, trace.CompareFailed(
+			"Wrong server subKind, was expecting %q, got %q",
+			types.SubKindOpenSSHNode,
```
Comment (marcoandredinis) [inline integrations/operator/controllers/resources/openssheiceserverv2_controller.go:47]:
```suggestion
			types.SubKindOpenSSHEICENode,
```

File: integrations/operator/controllers/resources/openssheiceserverv2_controller.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/gravitational/teleport/api/types"
+	resourcesv1 "github.com/gravitational/teleport/integrations/operator/apis/resources/v1"
+)
+
+// opensshEICEServerClient implements TeleportResourceClient and offers CRUD methods needed to reconcile provision tokens
+type opensshEICEServerClient struct {
+	teleportClient *client.Client
+}
+
+// Get gets the Teleport provision token of a given name
```
Comment (marcoandredinis) [inline integrations/operator/controllers/resources/openssheiceserverv2_controller.go:38]: Most godocs on this file, wrongly, reference the `provision token` resource

**Review by marcoandredinis — APPROVED** [review]

File: examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_opensshserversv2.yaml [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: OpenSSHServer resource definition v2 from Teleport
+            properties:
+              addr:
```
Comment (tigrato) [inline examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_opensshserversv2.yaml:38]: should we make this property mandatory or should we leave it to teleport?
This is a situation where validation hooks would be great

**Review by tigrato — APPROVED** [review]

File: examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_opensshserversv2.yaml (hunk 5 above)
Comment (hugoShaka) [reply examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_opensshserversv2.yaml:38]: I think teleport should reject an invalid value and we should keep the CRD validation to the minimum: enforcing a sane data structure.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**